### PR TITLE
Upgrade to xunit 2.3.0-beta4

### PR DESCRIPTION
**Pull request by natemcmaster**
Also, switch to using Directory.Build.props/targets and add xunit.runner.json

### CODE REVIEW

File: test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs [hunk 1]
```diff
@@ -211,11 +211,14 @@ public static TheoryData TryAddImplementationTypeData
...
-            var collection = new ServiceCollection();
-            collection.Add(ServiceDescriptor.Transient(expectedServiceType, expectedServiceType));
+            var collection = new ServiceCollection
+            {
+                ServiceDescriptor.Transient(expectedServiceType, expectedServiceType)
+            };
 
             // Act
             addAction(collection);
+            collection.TryAdd(ServiceDescriptor.Describe(expectedServiceType, expectedImplementationType, expectedLifetime));
```
Comment (pakrym) [inline test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs]: What does this fix?

Comment (natemcmaster) [inline test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs]: xunit1026: "unused theory parameters". The MemberData has items in it that aren't really needed for the test. We could either duplicate part of the memberdata into a colleciont that doesn't have these extra parameters, or use them.

Comment (pakrym) [inline test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs]: I don't like adding this like if accomplishes nothing but disabling a warning, which is false positive in this case.

Comment (natemcmaster) [inline test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs]: I think it fits with the description of the test: "TryAdd_WithType_DoesNotAddDuplicate". But you're welcome to refactor this test if it bothers you enough.

Comment (pakrym) [inline test/DI.Tests/ServiceCollectionServiceExtensionsTest.cs]: `addAction` already calls `TryAdd`. Please remove this line.

**Pull request merged**